Populate the RetryOpts of inspect and delete to avoid nil pointer der…

benjamint-bsquare opened the pull request:
…eference

A transient failure caused this nil pointer dereference:

Stack trace from the terraform-provider-skopeo_v0.0.4 plugin:

panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7674d6]

goroutine 66 [running]:
github.com/containers/common/pkg/retry.RetryIfNecessary({0x11ea898, 0xc000048078}, 0xc0002fd7f0, 0x0)
	github.com/containers/common@v0.46.1-0.20211026130826-7abfd453c86f/pkg/retry/retry.go:28 +0xb6
github.com/abergmeier/terraform-provider-skopeo/internal/skopeo.Inspect({0x11ea898, 0xc000048078}, {0xc0005705a0, 0x4d}, 0xc0002fd890)
	github.com/abergmeier/terraform-provider-skopeo/internal/skopeo/inspect.go:31 +0x116
github.com/abergmeier/terraform-provider-skopeo/internal/resources.exists(0x1058c8d, {0x6, 0x107da6e})
	github.com/abergmeier/terraform-provider-skopeo/internal/resources/copy_func.go:88 +0xa5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0006fc000, {0x11ea860, 0xc00054cd40}, 0xc00055cdd0, {0x0, 0x0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:615 +0x22a
...

benjamint-bsquare committed:
Populate the RetryOpts of inspect and delete to avoid nil pointer dereference

A transient failure caused this nil pointer dereference:

Stack trace from the terraform-provider-skopeo_v0.0.4 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7674d6]

goroutine 66 [running]:
github.com/containers/common/pkg/retry.RetryIfNecessary({0x11ea898, 0xc000048078}, 0xc0002fd7f0, 0x0)
	github.com/containers/common@v0.46.1-0.20211026130826-7abfd453c86f/pkg/retry/retry.go:28 +0xb6
github.com/abergmeier/terraform-provider-skopeo/internal/skopeo.Inspect({0x11ea898, 0xc000048078}, {0xc0005705a0, 0x4d}, 0xc0002fd890)
	github.com/abergmeier/terraform-provider-skopeo/internal/skopeo/inspect.go:31 +0x116
github.com/abergmeier/terraform-provider-skopeo/internal/resources.exists(0x1058c8d, {0x6, 0x107da6e})
	github.com/abergmeier/terraform-provider-skopeo/internal/resources/copy_func.go:88 +0xa5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0006fc000, {0x11ea860, 0xc00054cd40}, 0xc00055cdd0, {0x0, 0x0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:615 +0x22a
...

benjamint-bsquare committed:
Fork by replacing abergmeier with bsquare-corp

benjamint-bsquare committed:
Replaced module name abergmeier with bsquare-corp

benjamint-bsquare committed:
Replaced module name abergmeier with bsquare-corp v0.1.1

abergmeier commented:
Feel free to create you own fork of the provider.
I will not have any use for the provider for the time being.

benjamint-bsquare committed:
Replaced all references to abergmeier with bsquare-corp and removed the replace directive.

Now fully forked. Message from abergmeier:
 Feel free to create you own fork of the provider.
 I will not have any use for the provider for the time being.

benjamint-bsquare commented:
Dropping the PR as we have now forked the project
